### PR TITLE
feat(apm): Display the breakdown of the span operations

### PR DESCRIPTION
**Pull request by dashed**
## TODO

- [ ] ~add design~
- [x] some refactoring

### CODE REVIEW

File: src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          lookup[op] = 1;
+          return lookup;
+        }
+
+        lookup[op] = count + 1;
+
+        return lookup;
+      }
+
+      unknown += 1;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: Any reason you don't update the unknown key in the lookup?

File: src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+      return lookup;
+    }, {});
+
+    return {lookup: foo, unknown};
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: What's `foo`?

Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: foo is foo.

**Review by doralchan — APPROVED** [review]

File: src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import {
+  SentryTransactionEvent,
+  SpanEntry,
+  SpanType,
+} from 'app/components/events/interfaces/spans/types';
+import {TraceContextType} from 'app/components/events/interfaces/spans/traceView';
+
+type EventBreakdownType = {
+  ops: {name: string; count: number; percentage: number; duration: number}[];
+  unknown: {count: number; percentage: number; duration: number} | undefined;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: Should the `count, percentage, duration, name?` object have a named type?

File: src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    );
+
+    const spans: SpanType[] = get(spanEntry, 'data', []);
+
+    // track stats on spans with no operation
+    const spansWithNoOperation = {count: 0, duration: 0};
+
+    const numOfSpans = spans.length;
+
+    type AggregateType = {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: You name this type here, but couldn't it also be used in the EventBreakdown type?

File: src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    };
+
+    const traceContext: TraceContextType | undefined = get(event, 'contexts.trace');
+    if (traceContext) {
+      spans.push({
+        op: traceContext.op,
+        timestamp: event.endTimestamp,
+        start_timestamp: event.startTimestamp,
+        trace_id: '',
+        span_id: '',
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: Should the span_id and trace_id be using the data in the tracing context?

File: src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        return aggregate;
+      },
+      {}
+    );
+
+    const ops = Object.keys(aggregateByOp).map(opName => {
+      return {
+        name: opName,
+        count: aggregateByOp[opName].count,
+        percentage: aggregateByOp[opName].count / numOfSpans,
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: Shouldn't the percentage be by time not number of spans?

Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: Oh damn. I did it incorrectly.

File: src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx (hunk 4 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx]: `EventBreakdown` is array based whereas `AggregateType` is mapped based.

**Pull request merged**